### PR TITLE
Optimize performance

### LINKED ISSUE #12: Performance issues
FPS drops when menu is open.

### PR DESCRIPTION
**Pull request by kagikn**
Closes #12
No need to call `Enum.GetValues` every time...

### CODE REVIEW

**Comment by kagikn** [comment]
I should have noticed Tanjitsu had optimized performance for the shvdn3 branch in #68 before I made this PR...
OK, will make more commits for optimization.

**Comment by kagikn** [comment]
Disable certain controls would be better idea. The default interaction menu seems to disable only controls that prevent user from operating menus or cause slow mo.

**Comment by kagikn** [comment]
Btw, why don't you introduce [AppVeyor](https://www.appveyor.com)? This makes builds per commit, and you can let people using the builds with this. I use it for my scripts for SHVDN, and You can use it, too.

File: UIMenu.cs [hunk 1]
```diff
@@ -476,6 +482,37 @@ public void Clear()
...
 
+        private void DrawCalculations()
+        {
+            drawWidth = new Size(431 + WidthOffset, 107);
+
+            safe = GetSafezoneBounds();
+
+            backgroundSize = Size > MaxItemsOnScreen + 1 ? new Size(431 + WidthOffset, 38 * (MaxItemsOnScreen + 1)) : new Size(431 + WidthOffset, 38 * Size);
+
+            if (!String.IsNullOrWhiteSpace(MenuItems[_activeItem % (MenuItems.Count)].Description))
```
Comment (Guad) [inline UIMenu.cs]: That's a DivideByZeroException right here if MenuItems.Count is 0. You should move all item-size independent code to the top of the function and then return in the middle if there are no items.

**Comment by Guad** [comment]
As for AppVeyor, I doubt many people would use it. I'm waiting for SHV.NET 3.0 to release a final version but I have no idea when is that coming

**Comment by kagikn** [comment]
Now descriptions work well as before.
I have no idea when SHVDN v3 will come, too.
And, I'm sorry if I bothered you with AppVeyor.

**Comment by Tanjitsu** [comment]
@Guad with this being Master, i guess we will need to duplicate this into the shvdn3 once its ready ?

@kagikn at the start of the Draw Method. You can remove 

``` Point safe = GetSafezoneBounds(); ```

It is being done in DrawCalculations :)

**Comment by Guad** [comment]
@Tanjitsu Yeah I'll take care of that

**Comment by kagikn** [comment]
So, will you merge this PR if there is no problem?

**Comment by Guad** [comment]
Yeah, sorry for the delay

**Pull request merged**